contracts/src/tags/index.ts

```diff
@@ -0,0 +1,167 @@
+import { initContract } from '@ts-rest/core';
+import {
+  subDefaultTagQuerySchema,
+  subDefaultTagResponseSchema,
+  superDefaultTagResponseSchema,
+  superTagIdQuerySchema,
+  subTagResponseSchema,
+  tagsOfBookResponseSchema,
+  modifySuperTagBodySchema,
+  modifyTagResponseSchema,
+  incorrectTagFormatSchema,
+  alreadyExistTagSchema,
+  defaultTagCannotBeModifiedSchema,
+  modifySubTagBodySchema,
+  NoAuthorityToModifyTagSchema,
+  mergeTagsBodySchema,
+  invalidTagIdSchema,
+  createTagBodySchema,
+  duplicateTagSchema,
+  tagIdSchema,
+} from './schema';
+import {
+  bookInfoIdSchema,
+  bookInfoNotFoundSchema,
+  paginationQuerySchema,
+} from '../shared';
+
+const c = initContract();
+
+export const tagContract = c.router(
+  {
+    getSubDefault: {
+      method: 'GET',
+      path: '',
+      summary: '서브/디폴트 태그 정보를 검색한다.',
+      description: '서브/디폴트 태그 정보를 검색한다. 이는 태그 관리 페이지에서 사용한다.',
+      query: subDefaultTagQuerySchema,
+      responses: {
+        200: subDefaultTagResponseSchema,
+      },
+    },
+    getSuperDefaultForMain: {
+      method: 'GET',
+      path: '/main',
+      summary: '메인 페이지에서 사용할 태그 목록을 가져온다.',
+      description: '슈퍼 태그(노출되는 태그), 디폴트 태그(노출되지 않고 분류되지 않은 태그)를 랜덤한 순서로 가져온다. 이는 메인 페이지에서 사용된다.',
+      query: paginationQuerySchema.omit({ page: true }),
+      responses: {
+        200: superDefaultTagResponseSchema,
+      },
+    },
+    getSubOfSuperTag: {
+      method: 'GET',
+      path: '/{superTagId}/sub',
+      summary: '슈퍼 태그에 속한 서브 태그 목록을 가져온다.',
+      description: 'superTagId에 해당하는 슈퍼 태그에 속한 서브 태그 목록을 가져온다. 태그 병합 페이지에서 슈퍼 태그의 서브 태그를 가져올 때 사용한다.',
+      pathParams: superTagIdQuerySchema,
+      responses: {
+        200: subTagResponseSchema,
+      },
+    },
+    getSubOfSuperTagForAdmin: {
+      method: 'GET',
+      path: '/manage/{superTagId}/sub',
+      summary: '슈퍼 태그에 속한 서브 태그 목록을 가져온다.',
+      description: 'superTagId에 해당하는 슈퍼 태그에 속한 서브 태그 목록을 가져온다. 태그 관리 페이지에서 슈퍼 태그의 서브 태그를 가져올 때 사용한다.',
+      pathParams: superTagIdQuerySchema,
+      responses: {
+        200: subTagResponseSchema,
+      },
+    },
+    getTagsOfBook: {
+      method: 'GET',
+      path: '/{bookInfoId}',
+      summary: '도서에 등록된 슈퍼 태그, 디폴트 태그 목록을 가져온다.',
+      description: '슈퍼 태그(노출되는 태그), 디폴트 태그(노출되지 않고 분류되지 않은 태그)를 가져온다. 이는 도서 상세 페이지 및 태그 병합 페이지에서 사용된다.',
+      pathParams: bookInfoIdSchema,
+      responses: {
+        200: tagsOfBookResponseSchema,
+      },
+    },
+    modifySuperTag: {
+      method: 'PATCH',
+      path: '/super',
+      description: '슈퍼 태그를 수정한다.',
+      body: modifySuperTagBodySchema,
+      responses: {
+        200: modifyTagResponseSchema,
+        900: incorrectTagFormatSchema,
+        902: alreadyExistTagSchema,
+        906: defaultTagCannotBeModifiedSchema,
```

Comment:
```suggestion
        400: z.union([alreadyExistTagSchema, defaultTagCannotBeModifiedSchema]),
```
존재하는 HTTP 오류 코드를 사용하면 의도를 더 명확히 표현할 수 있을 것 같아요. 어떻게 생각하시나요?

참고 링크: https://softwareengineering.stackexchange.com/a/342896

Reply:
아 400이긴 한데 따로 400으로 빼긴 애매해서 그냥 커스텀으로 했더니 저렇게 하면 됐었네요..!

Reply:
혹시 다른 엔드포인트 반환값에도 900번대 대신 HTTP 오류 코드를 적용하는것에 대해 어떻게 생각하시나요?

Reply:
엄청 좋죠! 사실 커스텀 에러 코드를 E_001234 이런 형식이 아니라 숫자로 쓰는 것에 대해 회의적인 입장입니다